mllib/src/main/scala/org/apache/spark/mllib/tree/RandomForest.scala

```diff
@@ -244,7 +245,10 @@ private class RandomForest (
 
     // Delete any remaining checkpoints used for node Id cache.
     if (nodeIdCache.nonEmpty) {
-      nodeIdCache.get.deleteAllCheckpoints()
+      Try(nodeIdCache.get.deleteAllCheckpoints()) match {
+        case Failure(e) => logWarning(s"delete all chackpoints failed. Error reason: ${e.getMessage}")
```

Comment:
line too wide. Move logWarning to the next line.

Btw, I don't see how `Try` is useful here. Maybe we should just use

~~~
try {
  ...
} catch {
  case ...
  case ...
}
~~~

Java users would feel familiar with the code.